hook.go

```diff
@@ -62,43 +69,58 @@ func queryString(query interface{}) ([]byte, error) {
 	}
 }
 
-type queryProcessedHook func(*QueryProcessedEvent)
-
-// OnQueryProcessed calls the fn with QueryProcessedEvent
+// OnQueryEvent calls the interface with QueryEvent
 // when query is processed.
-func (db *DB) OnQueryProcessed(fn func(*QueryProcessedEvent)) {
-	db.queryProcessedHooks = append(db.queryProcessedHooks, fn)
+func (db *DB) OnQueryEvent(hook Hook) {
+	db.queryEventHooks = append(db.queryEventHooks, hook)
 }
 
-func (db *DB) queryProcessed(
+func (db *DB) queryStarted(
 	ormDB orm.DB,
-	start time.Time,
 	query interface{},
 	params []interface{},
 	attempt int,
-	res orm.Result,
-	err error,
-) {
-	if len(db.queryProcessedHooks) == 0 {
-		return
+) *QueryEvent {
+	if len(db.queryEventHooks) == 0 {
+		return nil
 	}
 
 	funcName, file, line := fileLine(2)
-	event := &QueryProcessedEvent{
-		StartTime: start,
-		Func:      funcName,
-		File:      file,
-		Line:      line,
+	event := &QueryEvent{
+		StartTime: time.Now(),
+		Func: funcName,
+		File: file,
+		Line: line,
```

Comment:
Can you please remove these params? I think it is reasonable to expect that users should save info they need in `Data` map.

Reply:
Done.